services/sidecar/src/simcore_service_sidecar/celery_configurator.py

```diff
@@ -7,217 +7,100 @@
-    )
-
-
-def shared_task_dispatch(
-    celery_request, user_id: str, project_id: str, node_id: Optional[str] = None
-) -> None:
-    log.info(
-        "Will dispatch to appropriate queue %s, %s, %s",
-        user_id,
-        project_id,
-        node_id,
-    )
-    next_task_nodes = wrap_async_call(
-        run_sidecar(celery_request.request.id, user_id, project_id, node_id)
-    )
-
-    # this needs to be done here since the tasks are created recursively and the state might not be upgraded yet
-    celery_request.update_state(state=states.SUCCESS)
-    if next_task_nodes:
-        for _node_id in next_task_nodes:
-            dispatch_comp_task(user_id, project_id, _node_id)
-
-
-def _on_task_failure_handler(
-    self, exc, task_id, args, kwargs, einfo
-):  # pylint: disable=unused-argument, too-many-arguments
-    log.error(
-        "Error while executing task %s with args=%s, kwargs=%s",
-        task_id,
-        args if args else "none",
-        pformat(kwargs) if kwargs else "none",
-    )
-
-
-def _on_task_success_handler(
-    self, retval, task_id, args, kwargs
-):  # pylint: disable=unused-argument
-    log.info(
-        "Task %s completed successfully with args=%s, kwargs=%s",
-        task_id,
-        args if args else "none",
-        pformat(kwargs) if kwargs else "none",
-    )
-
-
-def configure_cpu_mode() -> Tuple[RabbitConfig, Celery]:
-    """Will configure and return a celery app targetting CPU mode nodes."""
-    log.info("Initializing celery app...")
-    app = _celery_app_cpu
-
-    # pylint: disable=unused-variable,unused-argument
-    @app.task(
-        name="comp.task",
+def define_celery_task(app: Celery, name: str) -> None:
+    task = app.task(
+        name=name,
+        base=AbortableTask,
         bind=True,
         autoretry_for=(Exception,),
         retry_kwargs={"max_retries": 3, "countdown": 2},
-        on_failure=_on_task_failure_handler,
-        on_success=_on_task_success_handler,
+        on_failure=on_task_failure_handler,
+        on_success=on_task_success_handler,
         track_started=True,
-    )
-    def entrypoint(
-        self, *, user_id: str, project_id: str, node_id: Optional[str] = None
-    ) -> None:
-        shared_task_dispatch(self, user_id, project_id, node_id)
-
-    @app.task(
-        name="comp.task.cpu",
-        bind=True,
-        autoretry_for=(Exception,),
-        retry_kwargs={"max_retries": 3, "countdown": 2},
-        on_failure=_on_task_failure_handler,
-        on_success=_on_task_success_handler,
-        track_started=True,
-    )
-    def pipeline(
-        self, user_id: str, project_id: str, node_id: Optional[str] = None
-    ) -> None:
-        shared_task_dispatch(self, user_id, project_id, node_id)
-
-    set_boot_mode(BootMode.CPU)
-    log.info("Initialized celery app in %s ", get_boot_mode())
-    return (_rabbit_config, app)
+    )(celery_adapter(app)(entrypoint))
```

Comment:
could you split this on 3 separate lines so the 3 different calls are more obvious? It will make it more readable.

Reply:
done